src/replica.rs

```diff
@@ -236,27 +248,26 @@ impl Replica {
 
     /// Step 3. Validation of DebitAgreementProof, and credit idempotency at credit destination.
     /// (Since this leads to a credit, there is no requirement on order.)
-    pub fn receive_propagated(
+    pub fn receive_propagated<F: FnOnce() -> Option<PublicKey>>(
         &self,
         debit_proof: &DebitAgreementProof,
+        f: F,
```

Comment:
`f` is a very obscure naming. I had no idea what this was until I drilled down to where it is executed, and there was a comment.